app/controllers/health_cards_controller.rb

```diff
@@ -19,12 +19,16 @@ def scan; end
 
   def qr_contents
     contents = JSON.parse(params[:qr_contents])
+    jws_payload = HealthCards::Chunking.get_payload_from_qr contents
+    bundle = FHIR.from_contents(jws_payload['vc']['credentialSubject']['fhirBundle'].to_json)
```

Comment:
The code to process the qr contents shouldn't live in the controller. Extracting the contents of a QR code is a piece of business logic that should be executable and testable outside the context of an http request/response.

Reply:
Moved this logic to helper